internal/service/componentdescriptor/extractor.go

```diff
@@ -0,0 +1,61 @@
+package componentdescriptor
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	containerregistryv1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+// FileExtractor is responsible for extracting a specific file from a container image layer.
+type FileExtractor struct {
+	// reads all data from the provided reader
+	readAll func(reader io.Reader) ([]byte, error)
```

Comment:
If I understand it correctly, mocking this reader is the reason why this struct is tested "private". How about changing to the same pattern as we introduced for the OCI repository? I.e., make this configurable via a functional options pattern, then the test could be put into a separate `_test` package.